src/charm.py

```diff
@@ -331,7 +329,7 @@ def _configure(self, _: EventBase) -> None:  # noqa: C901
             return
         if not (vault := self._get_active_vault_client()):
             return
-        self._configure_pki_secrets_engine()
+        self._configure_pki_secrets_engine(vault)
```

Comment:
We could probably combine `configure()` with `sync()` (I haven't tried it, but I don't see any reason we can't).

I left it as-is to not change the behaviour too much here, but if there is support for combining them let me know.